examples/torch_ddp_benchmark.py

```diff
@@ -0,0 +1,305 @@
+# Adapted from https://github.com/pytorch/pytorch/tree/main/benchmarks/distributed/ddp
```

Comment:
Can we move the YAML and this script to `examples/torch_ddp_benchmark/`?

Reply:
done.